src/NumericalAlgorithms/Interpolation/InitializeInterpolationTarget.hpp

```diff
@@ -3,21 +3,98 @@
 
 #pragma once
 
-#include <algorithm>
+#include <cstddef>
+#include <tuple>
+#include <unordered_set>
 
 #include "DataStructures/DataBox/DataBox.hpp"
 #include "DataStructures/DataBox/DataBoxTag.hpp"
-#include "DataStructures/Variables.hpp"
-#include "Domain/Tags.hpp"
-#include "NumericalAlgorithms/Interpolation/Tags.hpp"
-#include "Parallel/ConstGlobalCache.hpp"
+#include "Domain/Domain.hpp"
+#include "Domain/Tags.hpp"                             // IWYU pragma: keep
+#include "NumericalAlgorithms/Interpolation/Tags.hpp"  // IWYU pragma: keep
+#include "Utilities/Requires.hpp"
+#include "Utilities/TMPL.hpp"
 #include "Utilities/TaggedTuple.hpp"
+#include "Utilities/TypeTraits.hpp"
+
+/// \cond
+// IWYU pragma: no_forward_declare db::DataBox
+namespace Parallel {
+template <typename Metavariables>
+class ConstGlobalCache;
+}  // namespace Parallel
+namespace Tags {
+template <typename TagsList>
+struct Variables;
+}  // namespace Tags
+/// \endcond
 
 namespace intrp {
 
 /// Holds Actions for Interpolator and InterpolationTarget.
 namespace Actions {
 
+// The purpose of the functions and metafunctions in this
+// namespace is to allow InterpolationTarget::compute_target_points
+// to omit an initialize function and a initialization_tags
+// type alias if it doesn't add anything to the DataBox.
+namespace initialize_interpolation_target_detail {
+
+// Sets value to initialization_tags, or
+// to empty list if initialization_tags is not defined.
+template <typename T, typename = cpp17::void_t<>>
+struct initialization_tags {
+  using value = tmpl::list<>;
+};
+
+template <typename T>
+struct initialization_tags<
+    T, cpp17::void_t<typename T::compute_target_points::initialization_tags>> {
+  using value = typename T::compute_target_points::initialization_tags;
+};
+
+// Tests whether T::compute_target_points has a
+// initialization_tags member.
+template <typename T, typename = cpp17::void_t<>>
+struct has_initialization_tags : std::false_type {};
+
+template <typename T>
+struct has_initialization_tags<
+    T, cpp17::void_t<typename T::compute_target_points::initialization_tags>>
+    : std::true_type {};
+
+// Tests whether T::compute_target_points has a non-empty
+// initialization_tags member.
+template <typename T>
+constexpr bool has_non_empty_initialization_tags_v =
+    has_initialization_tags<T>::value and
```

Comment:
this probably isn't needed for the desired behavior.
and if not, `has_initialization_tags` probably isn't needed.